### ISSUE TITLE
Building a release test build is not autosigned with a debug key 

### ISSUE THREAD

**Issue by sv-ohorvath** [mozilla-mobile/focus-android#4975]
## Steps to reproduce
Trying to run UI tests on a release build locally, from the command line, using:
`gradlew app:connectedAndroidTest -PtestBuildType=release -PdisableOptimization -Pandroid.testInstrumentationRunnerArguments.package="org.mozilla.focus.activity"`

What I've tried so far:
- I added this to build.gradle, as on Fenix: https://github.com/mozilla-mobile/fenix/blob/master/app/build.gradle#L23
- I've put autosignReleaseWithDebugKey in local.Properties too
- signed the apk with:
`jarsigner -keystore KEYSTORE .\app\build\outputs\apk\androidTest\focus\release\app-focus-release-androidTest.apk KEYNAME`
and then checked it is signed with ` jarsigner -verify -verbose -certs `

### Expected behavior
The release build can be installed on a device and run the UI tests.
### Actual behavior
I received this error:
` com.android.builder.testing.api.DeviceException: com.android.ddmlib.InstallException: INSTALL_PARSE_FAILED_NO_CERTIFICATES: Package /data/app/vmdl282333771.tmp/base.apk has no certificates at entry AndroidManifest.xml `
### Device information

* Android device: ?
* Focus version: ?

**Comment by Mugurell** [comment]
Looked with Oana at this.
Regarding the steps tried: 
- `autosignReleaseWithDebugKey` is not used in Focus but can be easily added to sign all builds with the debug key.
- If ^^ would work as in Fenix there's no need to sign the apk again. But even if signing it manually, without a way to pass the location of an existing apk so that `/gradlew app:connectedAndroidTest` would not build it's own apk that signed apk would be overwritten with an (atm) unsigned one.

-------

After adding `signingConfig signingConfigs.debug` to `app/build.gradle -> defaultConfig` (just as a quick way to get tests running on local release builds 😅) there's a [`No adjust token defined for release build`](https://github.com/mozilla-mobile/focus-android/blob/29f3be4aeec7e09b53103bc7cdd14fa19501cad0/app/src/focusRelease/java/org/mozilla/focus/utils/AdjustHelper.java#L27) exception killing the app.

I think we can easily get past this by using something like:
```
android.testVariants.all { variant ->
    variant.connectedInstrumentTest.doFirst {
        println "Special configuration for instrumented tests:"
        println "\tAdding a default \"null\" Adjust token"
        buildConfigField 'String', 'ADJUST_TOKEN', 'null'
    }
}
```
But below that adjust token call there's a check for [if telemetry is enabled](https://github.com/mozilla-mobile/focus-android/blob/29f3be4aeec7e09b53103bc7cdd14fa19501cad0/app/src/focusRelease/java/org/mozilla/focus/utils/AdjustHelper.java#L30) which will return true for non debug builds of Focus.
So I was thinking, shouldn't we disable telemetry (and possibly other app features) on these test builds?
@pocmo @rpappalax ?

**Comment by Mugurell** [comment]
After "making the test work" we see errors like
```
org.mozilla.focus.activity.AddToHomescreenTest > noNameShortcutTest[Pixel 3 - 11] FAILED 
        java.lang.NoClassDefFoundError: Failed resolution of: Lorg/mozilla/focus/ext/ContextKt;
        at org.mozilla.focus.helpers.MainActivityTestRuleKt.updateFirstRun(MainActivityTestRule.kt:103)
```

Guessing an R8 issue.

**Comment by Mugurell** [comment]
Think I should just make a PR for these changes.

**Comment by mcomella** [comment]
@Mugurell Can this be closed now that https://github.com/mozilla-mobile/focus-android/pull/4988 landed?

**Comment by Mugurell** [comment]
> @Mugurell Can this be closed now that #4988 landed?

Just opened a new PR to correct autlinking for the new section in README - https://github.com/mozilla-mobile/focus-android/pull/5066 

If @sv-ohorvath also confirms these changes work this can probably be closed.

**Comment by sv-ohorvath** [comment]
Yes, they work fine, thank you.
